assets/js/modules/analytics-4/datastore/settings.js

```diff
@@ -122,6 +122,29 @@ export async function submitChanges( { select, dispatch } ) {
 				// eslint-disable-next-line sitekit/acronym-case
 				webdatastream.webStreamData.measurementId
 			);
+
+			if ( webDataStreamID ) {
+				const isEnhancedMeasurementEnabled = select(
+					CORE_FORMS
+				).getValue(
+					ENHANCED_MEASUREMENT_FORM,
+					ENHANCED_MEASUREMENT_ENABLED
+				);
+
+				await dispatch(
+					MODULES_ANALYTICS_4
+				).setEnhancedMeasurementStreamEnabled(
+					propertyID,
+					webDataStreamID,
+					isEnhancedMeasurementEnabled
+				);
+				await dispatch(
+					MODULES_ANALYTICS_4
+				).updateEnhancedMeasurementSettings(
+					propertyID,
+					webDataStreamID
+				);
+			}
```

Comment:
This chunk is unnecessary since the main bug was elsewhere, but also because it is already handled below. The above can also set `streamEnabled` to `false` which isn't something it should do (see comment below).